src/util/SimpleTimer.h

```diff
@@ -0,0 +1,156 @@
+#pragma once
+
+#include "util/ThreadAnnotations.h"
+#include <medida/counter.h>
+#include <medida/metrics_registry.h>
+
+namespace stellar
+{
+
+template <typename> constexpr bool isDuration = false;
+
+template <typename Rep, typename Period>
+constexpr bool isDuration<std::chrono::duration<Rep, Period>> = true;
+
+template <typename Duration> class SimpleTimerContext;
+
+// Simple replacement for medida timer that uses an accumulator and counter
+// while keeping track of the maximums. Names are based on the constructor with
+// a suffix of `sum`, `count`, or `max`. Timers can be expensive, so this class
+// replaces them with a subset of the functionality using Counters internally.
+template <typename Duration> class SimpleTimer
+{
+    static_assert(
+        isDuration<Duration>,
+        "SimpleTimer must be instantiated with a std::chrono::duration");
+    medida::Counter& mSum;
+    medida::Counter& mCount;
+    // Note that we use a counter for `mMax` so it gets displayed in the
+    // metrics, but it is only synced on `syncMax()` so that the value of the
+    // counter is clear.
+    medida::Counter& mMaxCounter;
+    std::int64_t mMax GUARDED_BY(mLock);
+
+    Mutex mLock;
+
+  public:
+    // Specify `windowSize` to auto-call `syncMax` on every `Update` that takes
+    // place at least `windowSize` since the last `syncMax`.
```

Comment:
I'm not sure what this comment means, given that there is no `windowSize` parameter.